### PR TITLE
Use transport parameter in ENHO+SFPF

### LINKED ISSUE #5208: objects: use provided Transport in deserialization of ENHO+SFPF
see https://github.com/abapGit/abapGit/pull/5199

#5206 contains the solution for a similar downport issue

### PR DESCRIPTION
**Pull request by flaiker**
Downport of #5199 with the added parameter from #5216

I could only check 740 not any lower.

Edit: Fixes #5208

### CODE REVIEW

**Comment by larshp** [comment]
#5208

**Comment by larshp** [comment]
@filak-sap can you help reviewing this change?

File: src/objects/zcl_abapgit_object_sfpf.clas.abap [hunk 1]
```diff
@@ -193,16 +193,20 @@ CLASS zcl_abapgit_object_sfpf IMPLEMENTATION.
...
     TRY.
-        lo_wb_form->delete( lv_name ).
+        TRY.
+            CALL METHOD cl_fp_wb_form=>('DELETE')
+              EXPORTING
+                i_name     = lv_name
+                i_ordernum = iv_transport
+                i_dark     = abap_true. " > 740
+          CATCH cx_sy_dyn_call_error.
+            cl_fp_wb_form=>delete( lv_name ).
```
Comment (mbtools) [inline src/objects/zcl_abapgit_object_sfpf.clas.abap:208]: `i_ordernum` already exists in 702

File: src/objects/zcl_abapgit_object_sfpf.clas.abap [hunk 2]
```diff
@@ -232,12 +236,33 @@ CLASS zcl_abapgit_object_sfpf IMPLEMENTATION.
...
         IF zif_abapgit_object~exists( ) = abap_true.
-          cl_fp_wb_form=>delete( lv_name ).
+          TRY.
+              CALL METHOD cl_fp_wb_form=>('DELETE')
+                EXPORTING
+                  i_name     = lv_name
+                  i_ordernum = iv_transport
+                  i_dark     = abap_true. " > 740
+            CATCH cx_sy_dyn_call_error.
+              cl_fp_wb_form=>delete( lv_name ).
```
Comment (mbtools) [inline src/objects/zcl_abapgit_object_sfpf.clas.abap:246]: `i_ordernum` already exists in 702

File: src/objects/zcl_abapgit_object_sfpf.clas.abap [hunk 3]
```diff
@@ -232,12 +236,33 @@ CLASS zcl_abapgit_object_sfpf IMPLEMENTATION.
...
+            CALL METHOD cl_fp_wb_form=>('CREATE')
+              EXPORTING
+                i_name     = lv_name
+                i_form     = li_form
+                i_ordernum = iv_transport
+                i_dark     = abap_true " > 740
+              RECEIVING
+                r_wb_form  = li_wb_object.
+          CATCH cx_sy_dyn_call_error.
+            li_wb_object = cl_fp_wb_form=>create( i_name = lv_name
```
Comment (mbtools) [inline src/objects/zcl_abapgit_object_sfpf.clas.abap:262]: `i_ordernum` already exists in 702

File: src/objects/zcl_abapgit_object_sfpf.clas.abap (hunk 1 above)
Comment (flaiker) [reply src/objects/zcl_abapgit_object_sfpf.clas.abap:208]: Great, I'll move these 3 down then

**Review by mbtools — APPROVED** [review]
👍

**Comment by jfilak** [comment]
Good job guys!

**Pull request merged**